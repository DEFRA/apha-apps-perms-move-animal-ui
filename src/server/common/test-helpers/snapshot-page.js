import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
