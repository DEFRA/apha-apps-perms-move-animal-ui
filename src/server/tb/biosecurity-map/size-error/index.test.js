import { describePageSnapshot } from '../../../common/test-helpers/snapshot-page.js'
import { SizeErrorPage } from './index.js'

const sectionKey = 'biosecurity-map'
const key = 'size-error'
const pageTitle = 'There was a problem uploading your biosecurity map'
const view = 'tb/biosecurity-map/size-error/index'
const pageUrl = '/tb/biosecurity-map/size-error'

describe('#SizeErrorPage', () => {
  let page

  beforeEach(() => {
    page = new SizeErrorPage()
  })

  it('should have correct url path', () => {
    expect(page.urlPath).toBe(pageUrl)
  })

  it('should have the correct sectionKey', () => {
    expect(page.sectionKey).toBe(sectionKey)
  })

  it('should have the correct key', () => {
    expect(page.key).toBe(key)
  })

  it('should have the correct title', () => {
    expect(page.pageTitle).toBe(pageTitle)
  })

  it('should have the correct view', () => {
    expect(page.view).toBe(view)
  })

  describePageSnapshot({
    describes: 'SizeErrorPage.content',
    it: 'should render expected response and content',
    pageUrl
  })
})
