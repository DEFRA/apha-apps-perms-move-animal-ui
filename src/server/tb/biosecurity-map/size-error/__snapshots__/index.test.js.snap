// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SizeErrorPage SizeErrorPage.content should render expected response and content 1`] = `
"
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <h1 class="govuk-heading-l">There was a problem uploading your biosecurity map</h1>
      <div class="govuk-body" data-testid="app-page-body">
        <p>
          You must email your biosecurity map to csc.tblicensing@apha.gov.uk
        </p>
        <p>In your email include your county parish holding (CPH) number.</p>
        <p>
          Your application cannot be processed until we receive your biosecurity
          map.
        </p>
      </div>
    </div>
  </div>
    <form method="POST" novalidate="">
      <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
          <input type="hidden" name="crumb" value="csrf-value">
          <input type="hidden" name="nextPage" value="">
            <button type="submit" class="govuk-button" data-module="govuk-button" id="continue-button">
  Continue
</button>
        </div>
      </div>
    </form>
        "
`;
